cupy/__init__.py

```diff
@@ -874,3 +859,30 @@ def show_config():
     """Prints the current runtime configuration to standard output."""
     _sys.stdout.write(str(_cupyx.get_runtime_info()))
     _sys.stdout.flush()
+
+
+if _sys.version_info >= (3, 7):
+    _deprecated_attrs = {
+        'int': (int, 'cupy.int_'),
+        'bool': (bool, 'cupy.bool_'),
+        'float': (float, 'cupy.float_'),
+        'complex': (complex, 'cupy.complex_'),
+    }
+
+    def __getattr__(name):
+        value = _deprecated_attrs.get(name)
+        if value is None:
+            raise AttributeError(
+                f"module 'cupy' has no attribute '{name}'")
```

Comment:
For `getattr(cupy, '\n')`, `getattr(cupy, "'")`, and etc,
```suggestion
                f"module 'cupy' has no attribute {name!r}")
```

I found this repr (`!r`) in [PEP 562](https://www.python.org/dev/peps/pep-0562/), [in NumPy](https://github.com/numpy/numpy/blob/168978df69efecb07943fd8ce3a7c95aab7122cb/numpy/__init__.py#L308-L309), and [in unittest](https://github.com/python/cpython/blob/94894dd45e4bad6efb27eac4497b24cdc18b3e2d/Lib/unittest/__init__.py#L95), even though the default behavior is similar to `f"'{name}'"`.

```
>>> getattr(sys, '\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'sys' has no attribute '
'
>>> getattr(numpy, '\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../numpy/__init__.py", line 303, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute '\n'
```